app/assets/javascripts/discourse/app/models/user.js

```diff
@@ -449,11 +449,13 @@ const User = RestModel.extend({
       .then((result) => {
         this.set("bio_excerpt", result.user.bio_excerpt);
         const userProps = getProperties(
-          this.user_option,
+          result.user.user_option || this.user_option,
```

Comment:
Will there ever be a case where the `PUT` request does not return the user with the user option? I feel like we should always be using the server's response in this case.